server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -485,7 +500,7 @@ public TimeValue timeout() {
 
                 @Override
                 public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
-                    listener.onResponse(new RestoreCompletionResponse(snapshot, restoreInfo));
+                    listener.onResponse(new RestoreCompletionResponse(uuid, snapshot, restoreInfo));
```

Comment:
Passing the restore operation `uuid` here so we can use it in the transport action to trigger the correct listener.